File: Risk/LineView.m
//
// This file is a part of Risk by Mike Ferris.
//

//#import "Risk.h"

//RCSID ("$Id: LineView.m,v 1.2 1997/12/15 07:43:55 nygard Exp $");

#import "LineView.h"

@implementation LineView
@synthesize lineWidth;

- (void) setLineWidth:(CGFloat)lw
{
    lineWidth = lw;
    [self display];
}

//----------------------------------------------------------------------

- (void) drawRect:(NSRect)rects
{
    CGFloat mp, begin, end;
    NSRect boundsRect = self.bounds;
	
    NSDrawWhiteBezel (boundsRect, boundsRect);
    mp = NSMidY(boundsRect);
    begin = boundsRect.origin.x + 2;
    end = boundsRect.origin.x + boundsRect.size.width - 2;

	[[NSColor blackColor] set];
	NSBezierPath *path = [NSBezierPath bezierPath];
	path.lineWidth = lineWidth;
	[path moveToPoint:NSMakePoint(begin, mp)];
	[path lineToPoint:NSMakePoint(end, mp)];
	[path closePath];
	[path stroke];
}

@end

File: Risk/LineView.swift
//
//  LineView.swift
//  Risk
//
//  Created by C.W. Betts on 11/2/19.
//  Copyright © 2019 C.W. Betts. All rights reserved.
//

import Cocoa

/// Provide a simple view to show the width of a line, for use when
/// changing the border width.
@IBDesignable
class LineView: NSView {
    
    @IBInspectable @objc var lineWidth: CGFloat = 1 {
        didSet {
            needsDisplay = true
        }
    }
    
    override func draw(_ dirtyRect: NSRect) {
        let boundsRect = self.bounds
        
        NSDrawWhiteBezel(boundsRect, boundsRect)
        let mp = boundsRect.midY
        let begin = boundsRect.origin.x + 2
        let end = boundsRect.origin.x + boundsRect.size.width - 2
        
        NSColor.black.set()
        let path = NSBezierPath()
        path.lineWidth = lineWidth
        path.move(to: NSPoint(x: begin, y: mp))
        path.line(to: NSPoint(x: end, y: mp))
        path.close()
        path.stroke()
    }
}
